### PR TITLE
Update AppBar MediaQuery documentation

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Explain why you should consider wrapping AppBar in a MediaQuery 

## Related Issues

https://github.com/flutter/flutter/issues/27848

## Tests

I added the following tests:

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
@feinstein I made this section less elaborated (than promised), fearing it could be information overload for most readers. Please let me know if find it ambiguous or not helpful!

**Comment by feinstein** [comment]
Reading it the first thing that crosses my mind is "why would the ambient `MediaQuery` change?

Maybe the details should be added (and pointed to) at the `MediaQuery` docs as it will be mostly related to it and you could go deeply there.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -67,17 +67,20 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
...
 /// a [flexibleSpace] widget is specified then it is stacked behind the toolbar
 /// and the bottom widget. The following diagram shows where each of these slots
 /// appears in the toolbar when the writing language is left-to-right (e.g.
 /// English):
 ///
+/// The [AppBar] insets its content based on the ambient [MediaQuery]'s padding,
+/// to avoid system UI intrusions. It's taken care of by [Scaffold] when used in
+/// the [Scaffold.appBar] property. When animating an [AppBar], unexpected
+/// [MediaQuery] changes (as is common in [Hero] animations) may cause the content
+/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget with consistent
```
Comment (feinstein) [inline packages/flutter/lib/src/material/app_bar.dart]: What does `consistent padding values` mean?

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/app_bar.dart]: `Wrap the [AppBar] in a [MediaQuery] widget, adjust its padding such that the animation is smooth`?

Comment (feinstein) [reply packages/flutter/lib/src/material/app_bar.dart]: This implies that there isn't a recommended practice, it varies from case to case, is this right?

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/app_bar.dart]: Yes that's right. For example for a bottom `AppBar`, the top padding is more likely the culprit of a broken animation.

**Comment by feinstein** [comment]
So I guess it's alright as the message conveys it's intent.

File: packages/flutter/lib/src/material/app_bar.dart (hunk 1 above)
Comment (feinstein) [reply packages/flutter/lib/src/material/app_bar.dart]: So I guess it's alright as the message conveys it's intent.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -67,17 +67,20 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
...
 /// a [flexibleSpace] widget is specified then it is stacked behind the toolbar
 /// and the bottom widget. The following diagram shows where each of these slots
 /// appears in the toolbar when the writing language is left-to-right (e.g.
 /// English):
 ///
+/// The [AppBar] insets its content based on the ambient [MediaQuery]'s padding,
+/// to avoid system UI intrusions. It's taken care of by [Scaffold] when used in
+/// the [Scaffold.appBar] property. When animating an [AppBar], unexpected
+/// [MediaQuery] changes (as is common in [Hero] animations) may cause the content
+/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget, adjust its
```
Comment (justinmc) [inline packages/flutter/lib/src/material/app_bar.dart:81]:
```suggestion
/// to suddenly jump. Wrap the [AppBar] in a [MediaQuery] widget, and adjust its
```

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Comment by LongCatIsLooong** [comment]
@googlebot I fixed it.

**Pull request merged**